core/digital-twin-lib/src/DAppContainer.ts

```diff
@@ -298,12 +311,34 @@ class DAppContainer extends Container {
     return plugin;
   }
 
-  async getPermissions(): Promise<ContainerShareConfig> {
-    return {
-      read: [],
-      readWrite: [],
-      ...await this.getContainerShareConfigForAccount(this.runtime.activeAccount),
+  /**
+   * Returns permissions object with the same structure as the plugin.
+   * Adds `type {read: true}` to permissions, because it's omitted in getContainerShareConfigForAccount() function.
+   */
+  async getPermissions(): Promise<Permissions> {
+    const {
+      read = [],
+      readWrite = [],
+      removeListEntries = [],
+    } = await this.getContainerShareConfigForAccount(this.runtime.activeAccount);
+
+    const permissions = {
+      type: {
+        read: true,
+        readWrite: false,
+        removeListEntries: false,
+      },
     };
+
+    read.concat(readWrite).forEach((item) => {
```

Comment:
Does this include duplicates then? Iterate over a Set to remove them?

Reply:
it shouldn't include any duplicates as long as the bcc is working properly.